cfgov/agreements/management/commands/_util.py

```diff
@@ -42,9 +49,13 @@ def save_agreement(agreements_zip, pdf_path, outfile,
     uri_hostname = 'https://files.consumerfinance.gov'
     s3_prefix = 'a/assets/credit-card-agreements/pdf/'
 
-    zipinfo = agreements_zip.getinfo(pdf_path)
+    if six.PY2:
+        zipinfo = agreements_zip.getinfo(pdf_path)
+    else:
+        zipinfo = agreements_zip.getinfo(force_text(pdf_path, 'cp437'))
```

Comment:
What is the context for the 'cp437' here?

Reply:
Will add a comment about this change. But in brief (or not?):

Unless a particular flag is set in the zip file, it will default to MS-DOS encoding, which is `cp437`, according to the [zip file spec](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT):

> D.1 The ZIP format has historically supported only the original IBM PC character 
encoding set, commonly referred to as IBM Code Page 437.  This limits storing 
file name characters to only those within the original MS-DOS range of values 
and does not properly support file names in other character encodings, or 
languages. To address this limitation, this specification will support the 
following change. 
> 
> D.2 If general purpose bit 11 is unset, the file name and comment should conform 
to the original ZIP character encoding.  If general purpose bit 11 is set, the 
filename and comment must support The Unicode Standard, Version 4.1.0 or 
greater using the character encoding form defined by the UTF-8 storage 
specification.  The Unicode Standard is published by the The Unicode
Consortium (www.unicode.org).  UTF-8 encoded data stored within ZIP files 
is expected to not include a byte order mark (BOM).

Reply:
I think I might finagle that change a little bit more, because I'm still not sure it makes sense with the cp1252 encoding right below it.

Reply:
I've refactored the file encoding handling. There are still a couple of places where I don't fully understand what Python 2 is doing, but once we move to Python 3 the encodings flow through the import in a sensible way.